src/data/fintype/basic.lean

```diff
@@ -1293,10 +1293,15 @@ by { ext, simp [finset.mem_powerset_len] }
   fintype.card {s : finset α // s.card = k} = nat.choose (fintype.card α) k :=
 by simp [fintype.subtype_card, finset.card_univ]
 
-@[simp] lemma set.to_finset_univ [fintype α] :
-  (set.univ : set α).to_finset = finset.univ :=
+/-- See note [implicit instance arguments].
+Can use `set.to_finset_univ` when rewriting in reverse. -/
+@[simp] lemma set.to_finset_univ' {hu : fintype (set.univ : set α)} [fintype α] :
```

Comment:
Why not
```suggestion
@[simp] lemma set.to_finset_univ' [fintype (set.univ : set α)] [fintype α] :
```

Reply:
Previously, @jcommelin asked why not use implicit arguments.

I'm now confused about how `simp` handles typeclass arguments. I wouldn't have expected both of these examples to work:
```lean
@[simp] lemma set.to_finset_univ' [hu : fintype (set.univ : set α)] [fintype α] :
  @set.to_finset _ (set.univ : set α) hu = finset.univ :=
by { ext, simp only [set.mem_univ, finset.mem_univ, set.mem_to_finset] }

example [hα : fintype α] :
  @set.to_finset _ (set.univ : set α) (subtype.fintype _) = finset.univ := by simp

example [hα : fintype α] :
  @set.to_finset _ (set.univ : set α) set.fintype_univ = finset.univ := by simp
```

Does the "implicit instance arguments" note apply at all to `simp`?

Reply:
Hmm, I also don't know what's going on.

But I would be happy to merge the PR as it is now.

Reply:
Based on experiments and what @Vierkantor said on Zulip about how simp works, it's now back to using a typeclass argument.